tests/sqlparser_mssql.rs

```diff
@@ -100,48 +100,52 @@ fn parse_mssql_delimited_identifiers() {
 
 #[test]
 fn parse_create_procedure() {
-    let sql = "CREATE OR ALTER PROCEDURE test (@foo INT, @bar VARCHAR(256)) AS BEGIN SELECT 1 END";
+    let sql = "CREATE OR ALTER PROCEDURE test (@foo INT, @bar VARCHAR(256)) AS BEGIN SELECT 1; END";
```

Comment:
I think the semicolon is coming from `format_statement_list` here: https://github.com/apache/datafusion-sqlparser-rs/blob/ac1c339666c68779ed7f20dc0fb3b7473b298f83/src/ast/mod.rs#L163-L165

Not sure if we need to change any of that in this PR. It seems odd that semicolons _weren't_ part of the canonical SQL previously 🤔

Reply:
I think this behavior is OK, we try to preserve roundtrip behavior where possible but I'm suspecting that in this case, the effort required to work around it would not be worth the return